### ISSUE TITLE
IndexError: index 0 is out of bounds for axis 0 with size 0

### ISSUE THREAD

**Issue by fbemm** [nanoporetech/medaka#10]
Commit 2f896a70af6ba3bdae313a67b2a9663e1288e6a8

I am getting the following error while polishing a 550Mb genome with about 60x coverage. The assembly has 1169 contigs with an N50 of about 3Mb. Gut fealing is, that the loading of the feature.hdf fails silently. The mapping bam file is 65G and the resulting feature.hdf only 104Mb. 

```
[17:14:55 - medaka.compress] Skipping sample contig_1034:1.0-1072.0 which has 1435 columns < min 10000.
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 247, in bam_to_sample
    logging.info('Processed {} (median depth {})'.format(encode_sample_name(sample), np.median(depth_array)))
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/common.py", line 171, in encode_sample_name
    p['major'][0] + 1, p['minor'][0],
IndexError: index 0 is out of bounds for axis 0 with size 0
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/bin/hp_compress", line 11, in <module>
    load_entry_point('medaka==0.3.0', 'console_scripts', 'hp_compress')()
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 650, in main
    args.func(args)
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 617, in choose_feature_func
    features(args)
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 404, in features
    overlap=args.chunk_ovlp)
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/common.py", line 215, in write_samples_to_hdf
    for s in samples:
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 330, in alphabet_filter
    for s in sample_gen:
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/common.py", line 514, in <genexpr>
    return (c for s in samples for c in chunk_sample(s, chunk_len=chunk_len, overlap=overlap))
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 345, in min_positions_filter
    for s in sample_gen:
  File "/lib/python3.5/site-packages/medaka-0.3.0-py3.5.egg/medaka/compress.py", line 398, in <genexpr>
    samples = (s for g in sample_gens for s in g)
  File "/lib/python3.5/multiprocessing/pool.py", line 731, in next
    raise value
IndexError: index 0 is out of bounds for axis 0 with size 0
```

Medaka started with the following after mapping:

```
Using TensorFlow backend.
{'max_hp_len': 1, 'is_compressed': False, 'ref_mode': None, 'with_depth': False, 'consensus_as_ref': False, 'log_min': None, 'normalise': 'total'}
[17:11:09 - root] FeatureEncoder options: 
max_hp_len: 1
is_compressed: False
ref_mode: None
with_depth: False
consensus_as_ref: False
log_min: None
normalise: total
[17:11:09 - root] Creating consensus features.
```

Any ideas where to start debugging?

**Comment by fbemm** [comment]
I started to the 0.4.0rc2. Looks like I am running into a different issue now. Eventually needs a separate issue. To some extend conensus calling seems to work and medaka returned the 1st 11Mb but quit with the following error. I guess this is simply related to the contig being to short. Should one remove <10kb contigs before?

```
[12:13:22 - medaka.features] Skipping sample contig_1034:1.0-1072.0 which has 1435 columns < min 10000.
Traceback (most recent call last):
  File "/medaka/venv/bin/medaka", line 11, in <module>
    load_entry_point('medaka==0.4.0rc2', 'console_scripts', 'medaka')()
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/medaka.py", line 123, in main
    args.func(args)
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/inference.py", line 490, in predict
    batch_size=args.batch_size, n_samples=n_samples,
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/inference.py", line 236, in run_prediction
    for data in batches:
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/common.py", line 597, in grouper
    batch.append(next(gen))
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/common.py", line 545, in __next__
    return next(self.it)
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/common.py", line 588, in chain_thread_safe
    yield next(gen)
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/features.py", line 444, in alphabet_filter
    for s in sample_gen:
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/common.py", line 504, in <genexpr>
    return (c for s in samples for c in chunk_sample(s, chunk_len=chunk_len, overlap=overlap))
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/features.py", line 459, in min_positions_filter
    for s in sample_gen:
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/features.py", line 517, in <genexpr>
    samples = (s for g in sample_gens for s in g)
  File "/medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/features.py", line 290, in bam_to_sample
    raise ValueError('All reads were excluded, check you dtypes')
ValueError: All reads were excluded, check you dtypes
```

**Comment by cjw85** [comment]
Hi Felix, thanks for giving the release candidate a go. This is essentially the same issue, just the error has been pushed to a different part of the code. We are deciding how to best handle these cases, medaka can polish short contigs though we've been quite conservative in the shortest allowed size. If you would like to get your longer contigs processed, by all means remove the short ones from your dataset.

We are continuing to develop this release.

**Comment by fbemm** [comment]
Hi, getting closer ;) Excluding contigs <10kb allowed running until stitch. Stitch ended with:

```
Running medaka stitch
[02:15:36 - root] Label decoding is:
0: A
1: C
2: G
3: T
4: *
Traceback (most recent call last):
  File "medaka/venv/bin/medaka", line 11, in <module>
    load_entry_point('medaka==0.4.0rc2', 'console_scripts', 'medaka')()
  File "medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/medaka.py", line 123, in main
    args.func(args)
  File "medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/stitch.py", line 68, in stitch
    joined = stitch_from_probs(args.inputs, ref_names=args.ref_names, model_yml=args.model_yml)
  File "medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/stitch.py", line 47, in stitch_from_probs
    end_1_ind, start_2_ind = get_sample_overlap(s1, s2)
  File "medaka/venv/lib/python3.6/site-packages/medaka-0.4.0rc2-py3.6-linux-x86_64.egg/medaka/common.py", line 366, in get_sample_overlap
    assert len(pos1_ovl) == len(pos2_ovl)
AssertionError
```

**Comment by fbemm** [comment]
That is not going to be easy. Could you reach out to me via [mail](felix.bemm@kws.com)?

**Comment by cjw85** [comment]
Issue resolved after offline discussions and v0.4.0.
